src/org/zaproxy/zap/extension/pscanrulesAlpha/PiiScanner.java

```diff
@@ -0,0 +1,167 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import net.htmlparser.jericho.Source;import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A scanner to passively scan for the presence of PII in response
+ * Currently only credit card numbers
+ * @author Michael Kruglos (@michaelkruglos)
+ * TODO Extract credit card code when moved to beta(?) to also be used by InformationDisclosureReferrerScanner
+ */
+public class PiiScanner extends PluginPassiveScanner {
+
+    /**
+     * Prefix for internationalised messages used by this rule
+     */
+    private static final String MESSAGE_PREFIX = "pscanalpha.piiscanner.";
+
+    private static final int PLUGIN_ID = 10062;
+
+    private PassiveScanThread parent = null;
+
+    private enum CreditCard {
+        AMERICAN_EXPRESS("American Express", "(?:3[47][0-9]{13})"),
+        DINERSCLUB("DinersClub", "(?:3(?:0[0-5]|[68][0-9])[0-9]{11})"),
+        DISCOVER("Discover", "(?:6(?:011|5[0-9]{2})(?:[0-9]{12}))"),
+        JCB("Jcb", "(?:(?:2131|1800|35\\d{3})\\d{11})"),
+        MASESTRO("Maestro", "(?:(?:5[0678]\\d\\d|6304|6390|67\\d\\d)\\d{8,15})"),
+        MASTERCARD("Mastercard", "(?:(?:5[1-5][0-9]{2}|222[1-9]|22[3-9][0-9]|2[3-6][0-9]{2}|27[01][0-9]|2720)[0-9]{12})"),
+        VISA("Visa", "(?:4[0-9]{12})(?:[0-9]{3})?");
+
+        private final String name;
+        private final Pattern pattern;
+
+        CreditCard(String name, String regex) {
+            this.name = name;
+            this.pattern = Pattern.compile(regex);
+        }
+
+        public Matcher matcher(String cc) {
+            return pattern.matcher(cc);
+        }
+
+        @Override
+        public String toString() {
+            return name;
+        }
+    }
+
+    public PiiScanner() {
+    }
+
+    @Override
+    public void setParent(PassiveScanThread parent) {
+        this.parent = parent;
+    }
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        String responseBody = msg.getResponseBody().toString();
+        List<String> candidates = getNumberSequences(responseBody);
+        for(String candidate : candidates) {
+            for (CreditCard cc : CreditCard.values()) {
+                Matcher matcher = cc.pattern.matcher(candidate);
```

Comment:
This could just call the method, `cc.matcher(candidate);`.

Reply:
Done